### PR TITLE
WT-10525 Minimal repro for insert stack race condition

### PR DESCRIPTION
**Pull request by korteland**
There were some comments on the previous PR worth addressing:

* Names of some variables were fixed
* The probability is still 7/8, that's likely going to change in another PR.
* I don't think it makes sense to change the other use of the skip probability - this is just estimating the list size for eviction scheduling, which isn't behaviour we're interested in for this ticket.
* Cut down the amount of test config to the bare minimum.
* Also fixed a bunch of Evergreen issues
* Fixed a few nitpicks with the Python script. I chose not to validate the input - it's ultimately a GDB helper and not something people we expect to work without a little familiarity with what it's doing.
* Reduced the task timeout to 3600 seconds (1 hour)

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/evergreen.yml [hunk 1]
```diff
@@ -3476,6 +3476,43 @@ tasks:
...
+      - func: "compile wiredtiger"
+        vars:
+          CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v4_gcc.cmake
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/bench/workgen/runner"
+          script: |
+            set -o errexit
+            set -o verbose
+            for i in $(seq 150); do
```
Comment (ruby-oujo) [inline test/evergreen.yml:3493]: Have we tested we can successfully run this many iterations without it timing out within the hour? 

As an additional test, we would not want these tests to time out on ordinary runs.

File: test/evergreen.yml [hunk 2]
```diff
@@ -3476,6 +3476,43 @@ tasks:
...
+        vars:
+          NONSTANDALONE: -DWT_STANDALONE_BUILD=0
+          CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v4_gcc.cmake
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/bench/workgen/runner"
+          script: |
+            set -o errexit
+            set -o verbose
+            for i in $(seq 150); do
```
Comment (ruby-oujo) [inline test/evergreen.yml:3512]: as above.

File: test/evergreen.yml (hunk 1 above)
Comment (korteland) [reply test/evergreen.yml]: Thanks, good call - 150 was much too high. I've done some testing and ~5 was the highest that would reliably run, so I've pushed that in 99b4dc5.

**Review by ruby-oujo — APPROVED** [review]
LGTM :)

**Comment by korteland** [comment]
evergreen merge

**Pull request merged**